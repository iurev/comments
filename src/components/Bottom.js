import React from 'react';
import styled from 'styled-components'
import { Heart } from 'styled-icons/boxicons-solid/Heart'
import { Comment } from 'styled-icons/fa-solid/Comment'
import Ink from 'react-ink'

const Root = styled.div`
  display: flex;
  align-items: center;
  flex-wrap: wrap;
`

const Text = styled.div`
  line-height: 12px;
  font-size: 12px;
  color: ${p => p.theme.lightFontColor};
`

const Time = styled(Text)`
  margin-right: 10px;
`

const Stat = styled.div`
  display: flex;
  align-items: center;
  margin-right: 10px;
`

const StyledHeart = styled(Heart)`
  height: 20px;
  margin-right: 5px;
  path {
    fill: ${p => p.theme.lightFontColor};
  }
`

const StyledComment = styled(Comment)`
  height: 18px;
  margin-right: 5px;
  path {
    fill: ${p => p.theme.lightFontColor};
  }
`

const CommentButton = styled.button`
  position: relative;
  appearance: none;
  line-height: 20px;
  font-size: 12px;
  color: ${p => p.theme.lightFontColor};
  border: 0;
  padding: 0 5px;
  margin: 0;
  user-select: none;
  cursor: pointer;
  transition: color ease-out 0.3s;
  outline: none;

  &:hover, &:active {
    color: ${p => p.theme.mediumFontColor};
  }
`

const Bottom = ({ voteCount, answerCount, closed, time }) => (
  <Root>
    <Stat>
      <StyledHeart />
      <Text>{voteCount}</Text>
    </Stat>
    <Stat>
      <StyledComment />
      <Text>{answerCount}</Text>
    </Stat>
    <Time>vor {time}</Time>
    {!closed && <CommentButton><Ink />Kommentieren</CommentButton>}
  </Root>
);

export default Bottom;
