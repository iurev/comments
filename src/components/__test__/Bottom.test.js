import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import Bottom from '../Bottom';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Bottom />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('with all attrs', () => {
  const tree = renderer
    .create(<Bottom
      voteCount={1}
      answerCount={1}
      time="5 mins"
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('closed', () => {
  const tree = renderer
    .create(<Bottom
      voteCount={1}
      answerCount={1}
      time="5 mins"
      closed
    />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
