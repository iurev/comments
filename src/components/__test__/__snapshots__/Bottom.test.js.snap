// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`closed 1`] = `
<div
  className="sc-bdVaJa hKAetY"
>
  <div
    className="sc-bxivhb ePJekQ"
  >
    <svg
      aria-hidden="true"
      className="sc-ifAKCX hZJeFP StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20.205 4.791a5.938 5.938 0 0 0-4.209-1.754A5.906 5.906 0 0 0 12 4.595a5.904 5.904 0 0 0-3.996-1.558 5.942 5.942 0 0 0-4.213 1.758c-2.353 2.363-2.352 6.059.002 8.412L12 21.414l8.207-8.207c2.354-2.353 2.355-6.049-.002-8.416z"
      />
    </svg>
    <div
      className="sc-bwzfXH kPUBpl"
    >
      1
    </div>
  </div>
  <div
    className="sc-bxivhb ePJekQ"
  >
    <svg
      aria-hidden="true"
      className="sc-EHOje dfmlkX StyledIconBase-sc-bdy9j4 cCvWXa"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
    >
      <path
        d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32z"
        fill="currentColor"
      />
    </svg>
    <div
      className="sc-bwzfXH kPUBpl"
    >
      1
    </div>
  </div>
  <div
    className="sc-bwzfXH sc-htpNat dpzsFT"
  >
    vor 
    5 mins
  </div>
</div>
`;

exports[`with all attrs 1`] = `
<div
  className="sc-bdVaJa hKAetY"
>
  <div
    className="sc-bxivhb ePJekQ"
  >
    <svg
      aria-hidden="true"
      className="sc-ifAKCX hZJeFP StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20.205 4.791a5.938 5.938 0 0 0-4.209-1.754A5.906 5.906 0 0 0 12 4.595a5.904 5.904 0 0 0-3.996-1.558 5.942 5.942 0 0 0-4.213 1.758c-2.353 2.363-2.352 6.059.002 8.412L12 21.414l8.207-8.207c2.354-2.353 2.355-6.049-.002-8.416z"
      />
    </svg>
    <div
      className="sc-bwzfXH kPUBpl"
    >
      1
    </div>
  </div>
  <div
    className="sc-bxivhb ePJekQ"
  >
    <svg
      aria-hidden="true"
      className="sc-EHOje dfmlkX StyledIconBase-sc-bdy9j4 cCvWXa"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
    >
      <path
        d="M256 32C114.6 32 0 125.1 0 240c0 49.6 21.4 95 57 130.7C44.5 421.1 2.7 466 2.2 466.5c-2.2 2.3-2.8 5.7-1.5 8.7S4.8 480 8 480c66.3 0 116-31.8 140.6-51.4 32.7 12.3 69 19.4 107.4 19.4 141.4 0 256-93.1 256-208S397.4 32 256 32z"
        fill="currentColor"
      />
    </svg>
    <div
      className="sc-bwzfXH kPUBpl"
    >
      1
    </div>
  </div>
  <div
    className="sc-bwzfXH sc-htpNat dpzsFT"
  >
    vor 
    5 mins
  </div>
  <button
    className="sc-bZQynM jCrMHg"
  >
    <canvas
      height={0}
      onDragOver={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      style={
        Object {
          "borderRadius": "inherit",
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      width={0}
    />
    Kommentieren
  </button>
</div>
`;
